### PR TITLE
Photon ionization correction on S2

### PR DESCRIPTION
**Pull request by xzh19980906**
## What does the code in this PR do / what does it improve?
This PR implements the corrections on S2 due to photon ionization. Detailed study can be found in [the note](https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenonnt:zihao:sr1_s2aft_photonionization_correction).

## Can you briefly describe how it works?
It scales cs2 bottom based on the model and corrects cs2 aft accordingly.

### CODE REVIEW

**Review by jingqiangye — CHANGES_REQUESTED** [review]
Hi @xzh19980906 , thanks for the PR. The performance is good, see this [note](https://xe1t-wiki.lngs.infn.it/doku.php?id=jingqiang:straxen_review:1247).

However, I think we should not merge this PR until https://github.com/XENONnT/corrections/issues/221 is solved.

**Review by jingqiangye — APPROVED** [review]
Thanks @xzh19980906 for the update. The performance is satisfying, see [here](https://xe1t-wiki.lngs.infn.it/doku.php?id=jingqiang:straxen_review:1247).

@dachengx any idea why the uni tests failed?

**Comment by dachengx** [comment]
> @dachengx any idea why the uni tests failed?

Please just ignore it.

**Comment by shenyangshi** [comment]
I think we are now overwriting cs2_area_fraction_top, cs2 and cs2_bottom, maybe we can create extra fields before this correction such as cs2_wo_aftcorr, cs2_area_fraction_top_wo_aftcorr,  cs2_bottom_wo_aftcorr so we can know what's the raw fields in the future.

File: straxen/plugins/events/corrected_areas.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class CorrectedAreas(strax.Plugin):
         cs2_top and cs2_bottom are corrected by the corresponding maps,
         and cs2 is the sum of the two.
     """
-    __version__ = '0.3.0'
+    __version__ = '0.3.1'
```
Comment (dachengx) [inline straxen/plugins/events/corrected_areas.py:24]: Please update this to a minor bump. I think PI is a pretty large change.

**Comment by xzh19980906** [comment]
> I think we are now overwriting cs2_area_fraction_top, cs2 and cs2_bottom, maybe we can create extra fields before this correction such as cs2_wo_aftcorr, cs2_area_fraction_top_wo_aftcorr, cs2_bottom_wo_aftcorr so we can know what's the raw fields in the future.

done. I think `'_wo_picorr'` is better

**Review by shenyangshi — APPROVED** [review]
Looks good to me. Sometime in the future we should totally rewrite this corrected_areas :)

**Comment by jingqiangye** [comment]
Nice thanks guys!

**Comment by dachengx** [comment]
In principle, we should also change the `PeakCorrectedAreas` plugin: https://github.com/XENONnT/straxen/blob/2e9eaaa7e146e8a19573849293aa796ac46630ee/straxen/plugins/peaks/peak_corrections.py#L10.

**Pull request merged**